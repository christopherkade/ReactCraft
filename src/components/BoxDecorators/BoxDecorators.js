import React from "react"
import styled from "styled-components"

const TopLeft = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  width: 0.25rem;
  height: 0.25rem;
  background-color: #8C8C8C;
  z-index: 100;
  transform: rotate(45deg) translate(-10px);
  border: 2px ridge #8C8C8C;
`

const TopRight = styled.div`
  position: absolute;
  top: 0;
  right: 0;
  width: 0.25rem;
  height: 0.25rem;
  background-color: #8C8C8C;
  z-index: 100;
  transform: rotate(45deg) translateY(-10px);
  border: 2px ridge #8C8C8C;
`

const BottomRight = styled.div`
  position: absolute;
  bottom: 0;
  right: 0;
  width: 0.25rem;
  height: 0.25rem;
  background-color: #8C8C8C;
  z-index: 100;
  transform: rotate(45deg) translate(10px);
  border: 2px ridge #8C8C8C;
`

const BottomLeft = styled.div`
  position: absolute;
  bottom: 0;
  left: 0;
  width: 0.25rem;
  height: 0.25rem;
  background-color: #8C8C8C;
  z-index: 100;
  transform: rotate(45deg) translateY(10px);
  border: 2px ridge #8C8C8C;
`

const BoxDecorators = () => {
  return (
    <>
      <TopLeft />
      <TopRight />
      <BottomRight />
      <BottomLeft />
    </>
  )
}

export default BoxDecorators
