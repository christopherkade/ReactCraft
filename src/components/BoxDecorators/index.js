export { default as BoxDecorators } from "./BoxDecorators"
