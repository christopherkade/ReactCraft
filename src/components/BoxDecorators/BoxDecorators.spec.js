import React from 'react';
import BoxDecorators from './BoxDecorators';
import { render } from 'react-testing-library';

describe("<BoxDecorator />", () => {
  describe('Snapshots', () => {
    it("should match snapshot", () => {
      const { container } = render(<BoxDecorators />);
      expect(container).toMatchSnapshot();
    });
  });

  describe("Content validation", () => {
    it("should render 4 decorators", () => {
      const { container } = render(<BoxDecorators />)
      expect(container.querySelectorAll("div").length).toBe(4)
    })
  })
});
