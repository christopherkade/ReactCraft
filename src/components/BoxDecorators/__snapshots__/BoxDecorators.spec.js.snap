// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BoxDecorator /> Snapshots should match snapshot 1`] = `
<div>
  <div
    class="sc-bdVaJa fwHBtD"
  />
  <div
    class="sc-bwzfXH dRmJHm"
  />
  <div
    class="sc-htpNat iaYGmT"
  />
  <div
    class="sc-bxivhb fnUviR"
  />
</div>
`;
